client/syntaxes/bitbake.tmLanguage.json

```diff
@@ -1,270 +1,203 @@
-            },
-            "5": {
-              "name": "punctuation.definition.python-function.bb"
-            }
-          },
-          "end": "(\\})",
-          "endCaptures": {
-            "1": {
-              "name": "punctuation.definition.python-function.bb"
-            }
-          },
-          "patterns": [
-            {
-              "include": "source.python"
-            }
-          ]
+            "include": "#variable-reference"
         },
         {
-          "begin": "^\\s*(fakeroot)?\\s*(def)\\s*([a-zA-Z0-9-_:]+)*\\s*(\\([a-zA-Z_\\,\\ ]*\\))?\\s*(\\:)",
-          "beginCaptures": {
-            "1": {
-              "name": "keyword.control.fakeroot.bb"
-            },
-            "2": {
-              "name": "storage.type.function.python.bb"
-            },
-            "3": {
-              "name": "entity.name.function.python.bb"
-            },
-            "5": {
-              "name": "punctuation.definition.python-function.bb"
-            }
-          },
-          "end": "(?=^\\S)",
-          "patterns": [
-            {
-              "include": "source.python"
-            }
-          ]
+            "include": "#functions"
         },
         {
-          "begin": "^\\s*(fakeroot)?\\s*([a-zA-Z0-9-_:]+)\\s*(\\(\\))?\\s*(\\{)",
-          "beginCaptures": {
-            "1": {
-              "name": "keyword.control.fakeroot.bb"
-            },
-            "2": {
-              "name": "entity.name.function.shell.bb"
-            },
-            "4": {
-              "name": "punctuation.definition.shell-function.bb"
-            }
-          },
-          "end": "(\\})",
-          "endCaptures": {
-            "1": {
-              "name": "punctuation.definition.shell-function.bb"
-            }
-          },
-          "patterns": [
-            {
-              "include": "source.shell"
-            }
-          ]
-        }
-      ]
-    },
-    "variable-assignment": {
-      "begin": "(^\\s*|\\G\\s*)([^\\s]+)\\s*(=|\\?=|\\?\\?=|:=|\\+=|=\\+|\\.=|=\\.)\\s*",
-      "beginCaptures": {
-        "2": {
-          "name": "variable.other.bb"
+            "include": "#variable-name"
         },
-        "3": {
-          "name": "keyword.operator.assignment.bb"
-        }
-      },
-      "end": "\\n",
-      "patterns": [
         {
-          "match": "\\\\\\n",
-          "name": "constant.character.escape.continuation.bb"
+            "include": "#operator"
         },
         {
-          "include": "#comment"
+            "include": "#function-content"
+        }
+    ],
+    "repository": {
+        "keywords":{
+            "patterns": [
+                {
+                    "match": "^\\s*(include|require|inherit|export|echo|if|fi|else|return|EXPORT_FUNCTIONS)(?=\\s+|\\s*:)",
+                    "captures": {
+                        "1": {
+                            "name": "keyword.control.bb"
+                        }
+                    }
+                },
+                {
```

Comment:
Depending on how you want to handle the behaviour with the ":" after the keywords, it might be possible to remove this object and add the "fakeroot" keyword to the list at line 44